bindings/src/main/java/io/pravega/storage/azure/AzureChunkStorage.java

```diff
@@ -0,0 +1,210 @@
+/**
+ * Copyright Pravega Authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.pravega.storage.azure;
+
+import com.azure.storage.blob.models.BlobErrorCode;
+import com.azure.storage.blob.models.BlobStorageException;
+import com.google.common.base.Preconditions;
+import io.pravega.common.io.StreamHelpers;
+import io.pravega.segmentstore.storage.chunklayer.BaseChunkStorage;
+import io.pravega.segmentstore.storage.chunklayer.ChunkStorageException;
+import io.pravega.segmentstore.storage.chunklayer.ChunkHandle;
+import io.pravega.segmentstore.storage.chunklayer.ChunkNotFoundException;
+import io.pravega.segmentstore.storage.chunklayer.ConcatArgument;
+import io.pravega.segmentstore.storage.chunklayer.InvalidOffsetException;
+import io.pravega.segmentstore.storage.chunklayer.ChunkAlreadyExistsException;
+import io.pravega.segmentstore.storage.chunklayer.ChunkInfo;
+import lombok.SneakyThrows;
+import lombok.val;
+
+import java.io.BufferedInputStream;
+import java.io.InputStream;
+import java.util.concurrent.Executor;
+import java.util.concurrent.atomic.AtomicBoolean;
+
+
+public class AzureChunkStorage extends BaseChunkStorage {
+    private final AzureStorageConfig config;
+    private final AzureClient client;
+    private final boolean shouldClose;
+    private final AtomicBoolean closed;
+    private final boolean supportsAppend;
+
+    public AzureChunkStorage(AzureClient client, AzureStorageConfig config, Executor executor, boolean supportsAppend, boolean shouldClose) {
+        super(executor);
+        this.config = Preconditions.checkNotNull(config, "config");
+        this.client = Preconditions.checkNotNull(client, "client");
+        this.closed = new AtomicBoolean(false);
+        this.shouldClose = shouldClose;
+        this.supportsAppend = supportsAppend;
+
+    }
+
+    @Override
+    public boolean supportsTruncation() {
+        return false;
+    }
+
+    @Override
+    public boolean supportsAppend() {
+        return supportsAppend;
+    }
+
+    @Override
+    public boolean supportsConcat() {
+        return false;
+    }
+
+    @Override
+    protected ChunkInfo doGetInfo(String chunkName) throws ChunkStorageException {
+        try {
+            val blobProperties = client.getBlobProperties(getObjectPath(chunkName));
```

Comment:
If `client` is an attribute, please use `this.client` when using it (as you did below).